hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMultiMapProxy.java

```diff
@@ -91,6 +105,98 @@ public ClientMultiMapProxy(String serviceName, String name, ClientContext contex
         super(serviceName, name, context);
     }
 
+    @Override
+    public CompletionStage<Void> putAllAsync(@Nonnull Map<? extends K, Collection<? extends V>> m) {
+        InternalCompletableFuture<Void> future = new InternalCompletableFuture<>();
+        Map<Data, Data> dataMap = new HashMap<>();
+
+        for (Map.Entry e : m.entrySet()) {
+            Collection<Data> dataCollection = CollectionUtil
+                    .objectToDataCollection(((Collection<? extends V>) e.getValue()),
+                            getSerializationService());
+
+            dataMap.put(toData(e.getKey()), toData(new DataCollection(dataCollection)));
+        }
+        putAllInternal(dataMap, future);
+        return future;
+    }
+
+    @Override
+    public CompletionStage<Void> putAllAsync(@Nonnull K key, Collection<? extends V> value) {
+        InternalCompletableFuture<Void> future = new InternalCompletableFuture<>();
+        Map<Data, Data> dataMap = new HashMap<>();
+
+        Collection<Data> dataCollection = CollectionUtil
+                .objectToDataCollection(value, getSerializationService());
+        dataMap.put(toData(key), toData(new DataCollection(dataCollection)));
```

Comment:
A collection of objects should not be converted to Data. For details see my comment on protocol pr.
https://github.com/hazelcast/hazelcast-client-protocol/pull/313#issuecomment-612937041

Note that this is the case for Client only. For the member side, I believe it is ok the use DataCollection. 
For the client, we have a protocol and it needs to be self descriptive. Details are on the protocol pr comment